pkgs/json_rpc_2/lib/src/client.dart

```diff
@@ -49,9 +51,14 @@ class Client {
   ///
   /// Note that the client won't begin listening to [channel] until
   /// [Client.listen] is called.
-  Client(StreamChannel<String> channel)
+  ///
+  /// If [generateId] is passed, it will be called to generate an ID for each
+  /// request. Defaults to an auto-incrementing `int`. The value returned must
```

Comment:
re: 'must be either an int or a string' - it that as per the json-rpc spec?

Reply:
Yes

Reply:
"An identifier established by the Client that MUST contain a String, Number, or NULL value if included. If it is not included it is assumed to be a notification. The value SHOULD normally not be Null [[1]](https://www.jsonrpc.org/specification#id1) and Numbers SHOULD NOT contain fractional parts [[2]](https://www.jsonrpc.org/specification#id2)"

I suppose we could actually allow `num` or `String`, even though decimals are very risky to use, its technically only a "SHOULD NOT". Thoughts?

Reply:
I was just verifying that this limitation is there for a reason; keeping it to ints and strings (via docs) makes sense.